pkgs/development/ruby-modules/bundler-env/default.nix

```diff
@@ -67,25 +76,56 @@ let
     if hasBundler then gems.bundler
     else defs.bundler.override (attrs: { inherit ruby; });
 
-  gems = lib.flip lib.mapAttrs configuredGemset (name: attrs:
-    buildRubyGem ((removeAttrs attrs ["source"]) // attrs.source // {
-      inherit ruby;
-      gemName = name;
-      gemPath = map (gemName: gems."${gemName}") (attrs.dependencies or []);
-    }));
+  pathDerivation = {
+    usesGemspec ? false, ...
+  }@attrs:
+    let
+      res = {
+          inherit usesGemspec;
+          type = "derivation";
+          name = attrs.gemName;
+          version = attrs.version;
+          outPath = attrs.path;
+          outputs = [ "out" ];
+          out = res;
+          outputName = "out";
+        };
+    in res;
```

Comment:
The problem being solved here is the use of gems that are on the local filesystem e.g. not in the nix store. So this wrapper is a kind of fake derivation to make that possible.

One thing I'm not sure how to do, but I think needs to be added is that these "derivations" need to taint their dependent - it can be used as a shell.nix, but there's no way it should be put into nixpkgs like this.

There's one exception here, and that's the mainGem, because its files (should) get copied into the nix store as part of gemfile-and-lockfile.